### ISSUE TITLE
Error: C stack usage  7969252 is too close to the limit

### ISSUE THREAD

**Issue by ramiromagno** [r-lib/testthat#812]
I am not sure why I am getting this error when running `devtools::test()`. Using testthat 2.0.0.

Here's the minimal example:

Function to be tested, `dummy()`:

```R
dummy <- function() {
  warning("A warning.")
}
```

Code for its testing:

```R
test_that("Test dummy", {
  expect_warning(dummy(), "A warning.")
})
```
Which generates:

```
Error: C stack usage  7969252 is too close to the limit
```

Am I doing something wrong? Or is it a `testthat` bug?

**Comment by ramiromagno** [comment]
It seems the problem had to do with me having redefined the function `quote()`... `testthat` uses this function... so when I renamed my `quote` function to something else then `devtools::test()` stop giving me the `C stack usage` error.

It would be nice if someone would check this hypothesis.

**Comment by ramiromagno** [comment]
@jimhester absolutely! :D It was not purpose though... btw, if you guys had used `base::quote` inside the `testthat` pkg, would it have not avoided this cross-fire?

**Comment by kevinushey** [comment]
There are a number of places where R internally used the version of `quote()` defined in the global environment, even when that should not be the case. For example:

```
> quote <- function(...) stop("ouch!")
> warning("oops")
Error in quote(NULL) : ouch!
```

AFAIK this will be fixed in R 3.6.0 but until then it's not advisable to override the definition of `quote()` with your own version.
